### PR TITLE
Fix for issue #483, original by fx-kirin

### LINKED ISSUE #483: #354 fix causes IndexError: string index out of range
## Classification:

Bug

## Reproducibility:

Just run this command through autokey.

``` python
system.exec_command("fcitx-remote -s **fcitx-keyboard-us")
```

## Version

AutoKey version: 0.95.10

Used GUI (Gtk, Qt, or both): 

If the problem is known to be present in more than one version, please list all of those.

Installed via: (PPA, pip3, …). PPA

Linux Distribution: Linux Mint 20

## Summary

#354 adds some commands related to this problem. It causes the result.

## Expected Results

Nothing happens.

## Actual Results

``` python
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/autokey/service.py", line 207, in __tryReleaseLock
    self.configManager.lock.release()
RuntimeError: release unlocked lock
2020-11-29 23:28:13,811 ERROR - service - Script error
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/autokey/service.py", line 485, in execute
    exec(script.code, scope)
  File "<string>", line 2, in <module>
  File "/usr/lib/python3/dist-packages/autokey/scripting.py", line 497, in exec_command
    if output[-1] == "\n":
IndexError: string index out of range
```

### PR DESCRIPTION
**Pull request by sebastiansam55**
Also swaps from striping trailing new line using indexes to `rstrip('\n')`.

See https://github.com/autokey/autokey/pull/486/files#r532656403.

Output of `subprocess` seems to always be at least a empty string: `''`, so checking length shouldn't be a problem, no more than the previous method, `endswith` which also (I think) assumes output to be a string.

Credit goes to https://github.com/fx-kirin for original fix.

### CODE REVIEW

**Comment by mvindiola1** [comment]
I ran into this error today. I wanted to add two comments that you might consider.

1. You do not really need the conditional because if you rstrip an empty string you get the empty string back. You could do it cleanly and more expressively in one line like this:
    `output = p.communicate()[0].rstrip('\n')`.

    As an aside, if you want to use python indexing, the correct way to test this would have been:
    `if t[-1:] == "\n":`

2. There are a couple other places where you could do the same thing if you wanted consistency :
https://github.com/autokey/autokey/blob/c2ebcbba75ce40ed080c78216730b93690cb9a21/lib/autokey/scripting/dialog_gtk.py#L42
https://github.com/autokey/autokey/blob/c2ebcbba75ce40ed080c78216730b93690cb9a21/lib/autokey/scripting/dialog_qt.py#L42
https://github.com/autokey/autokey/blob/c2ebcbba75ce40ed080c78216730b93690cb9a21/lib/autokey/scripting/window.py#L244

File: lib/autokey/scripting/system.py [hunk 1]
```diff
@@ -43,11 +43,10 @@ def exec_command(command, getOutput=True):
                     stdout=subprocess.PIPE,
                     universal_newlines=True) as p:
                 output = p.communicate()[0]
-                if output.endswith("\n"):
+                if len(output) > 0:
```
Comment (BlueDrink9) [inline lib/autokey/scripting/system.py]: Good advice @mvindiola1. @sebastiansam55 if you just remove this conditional, then @josephj11 should merge this change.

As an aside, I couldn't replicate this issue with automated tests, so if you can figure out how to, add the test to `autokey/tests/scripting_api/test_system.py`

**Review by BlueDrink9 — CHANGES_REQUESTED** [review]

File: lib/autokey/scripting/system.py (hunk 1 above)
Comment (BlueDrink9) [reply lib/autokey/scripting/system.py]: @sebastiansam55 can you quickly please replace lines 46-49 with

```python
# Most shell output has a new line at the end, which we don't want.
output = output.rstrip("\n")
```

Then repush and we can merge. This has cropped up as a bug elsewhere.

**Comment by BlueDrink9** [comment]
@josephj11 Should be good to merge

**Pull request merged**